packages/trpc/server/routers/viewer/teams.tsx

```diff
@@ -562,4 +616,217 @@ export const viewerTeamsRouter = createProtectedRouter()
         },
       });
     },
+  })
+  .query("validateTeamName", {
+    input: z.object({
+      name: z.string(),
+    }),
+    async resolve({ ctx, input }) {
+      const team = await ctx.prisma.team.findFirst({
+        where: {
+          name: input.name,
+        },
+      });
+
+      return !team;
+    },
+  })
+  .query("validateTeamSlug", {
+    input: z.object({
+      slug: z.string(),
+    }),
+    async resolve({ ctx, input }) {
+      const team = await ctx.prisma.team.findFirst({
+        where: {
+          slug: input.slug,
+        },
+      });
+
+      return !team;
+    },
+  })
+  .query("findUser", {
+    input: z.object({
+      emailOrUsername: z.string(),
+      role: z.object({
+        value: z.union([z.literal("OWNER"), z.literal("ADMIN"), z.literal("MEMBER")]),
+        label: z.string(),
+      }),
+    }),
+    async resolve({ ctx, input }) {
+      const { emailOrUsername } = input;
+      const user = await ctx.prisma.user.findFirst({
+        where: {
+          OR: [{ username: emailOrUsername }, { email: emailOrUsername }],
+        },
+        select: {
+          id: true,
+          name: true,
+          username: true,
+          avatar: true,
+          email: true,
+          locale: true,
+        },
+      });
+
+      if (!user && !/\b[a-z0-9-_.]+@[a-z0-9-_.]+(\.[a-z0-9]+)+/.test(emailOrUsername)) {
+        throw new TRPCError({
+          code: "NOT_FOUND",
+          message: "Could not find user",
+        });
+      }
+
+      return {
+        id: user?.id || undefined,
+        name: user?.name || "",
+        email: user?.email || emailOrUsername,
+        username: user?.username || "",
+        avatar: user?.avatar || "",
+        role: input.role.value,
+        locale: user?.locale || "en",
+      };
+    },
+  })
+  .query("getTeamPrices", {
+    async resolve() {
+      const teamPrices = getTeamPricing();
+      return teamPrices;
+    },
+  })
+  .mutation("createPaymentIntent", {
+    input: z.object({
+      teamName: z.string(),
+      billingFrequency: z.string(),
+      seats: z.number(),
+      customerId: z.string().optional(),
+      subscriptionId: z.string().optional(),
+    }),
+    async resolve({ ctx, input }) {
+      const { teamName, billingFrequency, seats, customerId, subscriptionId } = input;
+
+      // Check to see if customer already exists in DB from another session
+      let customer = await searchForTeamCustomer(teamName, ctx.user.email);
```

Comment:
If navigating between screens or coming back from another session, see if the customer with the same team name and email is already in Stripe. Should be unique since team names are unique.